internal/provider/resource_monitor_resource.go

```diff
@@ -0,0 +1,807 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/setvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/float64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/float64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var (
+	_ resource.Resource                = &ResourceMonitorResource{}
+	_ resource.ResourceWithImportState = &ResourceMonitorResource{}
+)
+
+func NewResourceMonitorResource() resource.Resource {
+	return &ResourceMonitorResource{}
+}
+
+type ResourceMonitorResource struct {
+	client *sdk.Client
+}
+
+type resourceMonitorModelV0 struct {
+	Name                     types.String  `tfsdk:"name"`
+	NotifyUsers              types.Set     `tfsdk:"notify_users"`
+	CreditQuota              types.Float64 `tfsdk:"credit_quota"`
+	Frequency                types.String  `tfsdk:"frequency"`
+	StartTimestamp           types.String  `tfsdk:"start_timestamp"`
+	EndTimestamp             types.String  `tfsdk:"end_timestamp"`
+	SuspendTrigger           types.Int64   `tfsdk:"suspend_trigger"`
+	SuspendTriggers          types.Set     `tfsdk:"suspend_triggers"`
+	SuspendImmediateTrigger  types.Int64   `tfsdk:"suspend_immediate_trigger"`
+	SuspendImmediateTriggers types.Set     `tfsdk:"suspend_immediate_triggers"`
+	NotifyTriggers           types.Set     `tfsdk:"notify_triggers"`
+	SetForAccount            types.Bool    `tfsdk:"set_for_account"`
+	Warehouses               types.Set     `tfsdk:"warehouses"`
+	Id                       types.String  `tfsdk:"id"`
+}
+
+func resourceMonitorSchemaV0() schema.Schema {
+	return schema.Schema{
+		Version: 0,
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
```

Comment:
is there any way to use `resourceMonitorModelV0` to not write everything here (or use this to generate the above struct)?

What I mean is that in the model above I have to say that the `Name` will be a `string` and that it uses the `"name"` key. Here I have to specify it again (`"name"`, `StringAttribute`).

It seems super-clunky. Aren't there any tools to make it less repetitive?

Reply:
Not from the documentation I have read. This was what I was saying, when I think there is stuff we could generate.